### PR TITLE
WebSocket API fixes

### PR DESCRIPTION
**Pull request by dogkisser**

### CODE REVIEW

**Comment by dogkisser** [comment]
let me cook on this one

**Comment by pupgray** [comment]
Drafted during cooking

**Comment by dogkisser** [comment]
i ordered pizza instead
Works perfectly in my testing :)

File: app/channels/link_channel.rb [hunk 1]
```diff
@@ -3,37 +3,35 @@ def subscribed
...
   end
 
   def unsubscribed
-    if connection&.watched_link
-      connection.watched_link.live_client_started_at = nil
-      connection.watched_link.save
+    if params[:id].present?
+      link = Link.find(params[:id])
+      if link
+        stop_stream_from "Link::#{params[:id]}"
```
Comment (pupgray) [inline app/channels/link_channel.rb:19]: So, in my testing, this works totally fine, but stop_stream_for might be a bit more rails-y

https://api.rubyonrails.org/v7.1.1/classes/ActionCable/Channel/Streams.html#method-i-stop_stream_for

File: app/channels/application_cable/connection.rb [hunk 2]
```diff
@@ -1,20 +1,20 @@
 module ApplicationCable
   class Connection < ActionCable::Connection::Base
-    attr_accessor :watched_link, :current_user
+    attr_accessor :watched_links, :current_user
 
     def connect
-      @watched_link = nil
+      @watched_links = []
```
Comment (pupgray) [inline app/channels/application_cable/connection.rb:6]: Ruby developer mindset, pluralizing like DHH having a dissociative psychotic event ❤️

**Review by pupgray — APPROVED** [review]
lgtm, some random bullshit suggestion

**Comment by pupgray** [comment]
Will merge this in a sec, deploying in parts

File: app/channels/link_channel.rb (hunk 1 above)
Comment (dogkisser) [reply app/channels/link_channel.rb:19]: Ah, like `stop_stream_for link`?

File: app/channels/application_cable/connection.rb (hunk 2 above)
Comment (dogkisser) [reply app/channels/application_cable/connection.rb:6]: I'm just impressed I could push a Link object into it and everything just works™️

You know the kind of memory implications that would have in Rust??

**Comment by pupgray** [comment]
Reheating it

**Pull request merged**